src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const currentState = { ...state };
+
+  for (const action of actions) {
+    const { extraData, type, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        // eslint-disable-next-line no-const-assign
+        for (const key in currentState) {
+          delete currentState[key];
```

Comment:
Directly deleting properties from 'currentState' is mutating the state object. Instead, you should create a new object for each action to adhere to the requirement of not mutating the original state or any previous state.